src/osal/lv_freertos.c

```diff
@@ -304,31 +325,43 @@ lv_result_t lv_thread_sync_signal(lv_thread_sync_t * pxCond)
 
 lv_result_t lv_thread_sync_delete(lv_thread_sync_t * pxCond)
 {
-#if USE_FREERTOS_TASK_NOTIFY
-    LV_UNUSED(pxCond);
-#else
+#if !USE_FREERTOS_TASK_NOTIFY
     /* Cleanup all resources used by the cond. */
     vSemaphoreDelete(pxCond->xCondWaitSemaphore);
     vSemaphoreDelete(pxCond->xSyncMutex);
     pxCond->ulWaitingThreads = 0;
     pxCond->xSyncSignal = pdFALSE;
-    pxCond->xIsInitialized = pdFALSE;
 #endif
+    pxCond->xIsInitialized = pdFALSE;
 
     return LV_RESULT_OK;
 }
 
 lv_result_t lv_thread_sync_signal_isr(lv_thread_sync_t * pxCond)
 {
     BaseType_t xHigherPriorityTaskWoken = pdFALSE;
-#if USE_FREERTOS_TASK_NOTIFY
-    /* Send a notification to the task waiting. */
-    vTaskNotifyGiveFromISR(pxCond->xTaskToNotify, &xHigherPriorityTaskWoken);
-    portYIELD_FROM_ISR(xHigherPriorityTaskWoken);
-#else
+
     /* If the cond is uninitialized, perform initialization. */
     prvCheckCondInit(pxCond);
 
+#if USE_FREERTOS_TASK_NOTIFY
+    _enter_critical();
```

Comment:
`_enter_critical()` does not work for ISR, it will hang.
You would have to add new defines:
```
#define _enter_critical_isr() taskENTER_CRITICAL_FROM_ISR();
```

Reply:
Very good catch! :open_mouth: In all of the environments I've seen so far, the same critical section API is used inside and outside of ISRs. I tested on ESP32-S3 and it worked for me. Maybe only some ports of FreeRTOS implement the ISR version with important extra steps.